src/test/java/net/sf/jabref/logic/importer/fileformat/BibtexImporterTest.java

```diff
@@ -121,4 +121,12 @@ public void testGetDescription() {
                         + "The metadata is not required to be read here, as this class is NOT called at --import.",
                 importer.getDescription());
     }
+
+    @Test
+    public void testRecognizesDatabaseID() throws Exception {
+        Path file = Paths.get(BibtexImporterTest.class.getResource("AutosavedSharedDatabase.bib").toURI());
```

Comment:
Could you please add the database id also to complex.bib. This test file should illustrate every feature Jabref has. Then also an additional `AutosavedSharedDb.bib` is no longer necessary.